docs/documentation.md

```diff
@@ -0,0 +1,430 @@
+`saleCut` only supports a single token receiver type and therefore beneficiaries of a `saleCut` can only receive the token type used for the purchase. To support different token types for saleCuts we recommend using the [FungibleTokenSwitchboard](https://github.com/onflow/flow-ft/blob/master/contracts/FungibleTokenSwitchboard.cdc) contract.
+
+***Note:*** We recommend that marketplaces honor creator royalties across the Flow ecosystem
+
+## Enabling marketplace commissions for NFT sales
+
+`NFTStorefrontV2` enables optional commissions on trades for marketplaces which require it as a condition to list a NFT for sale. Commission & commission receivers are set by the seller during initial listing creation. At time of purchase the commission amount is paid once only to the commission receiver matching the marketplace receiver address which facilitated the sale. When commission is set but no commission receivers are specified, the commission is paid to [TBD].
+
+For NFT listings in marketplaces which don't require commission, commission receivers can be set as `nil`. Setting the buyer of the NFT and `commissionRecipient` to the same has the effect of applying a discount for the buyer.
+
+![scenario_2](https://user-images.githubusercontent.com/14581509/190966499-c176203f-b6a6-4422-860f-1bf6f2bcdbb6.png).
+
+## APIs & Events offered by NFTStorefrontV2
+
+## Resource Interface `ListingPublic`
+
+```cadence
+resource interface ListingPublic {
+    pub fun borrowNFT(): &NonFungibleToken.NFT?
+    pub fun purchase(
+          payment: @FungibleToken.Vault, 
+          commissionRecipient: Capability<&{FungibleToken.Receiver}>?,
+      ): @NonFungibleToken.NFT
+    pub fun getDetails(): ListingDetail
+    pub fun getAllowedCommissionReceivers(): [Capability<&{FungibleToken.Receiver}>]?
+}
+```
+An interface providing a useful public interface to a Listing.
+
+### Functions
+
+**fun `borrowNFT()`**
+
+```cadence
+fun borrowNFT(): &NonFungibleToken.NFT?
+```
+This will assert in the same way as the NFT standard borrowNFT()
+if the NFT is absent, for example if it has been sold via another listing.
+
+---
+
+**fun `purchase()`**
+
+```cadence
+fun purchase(payment FungibleToken.Vault, commissionRecipient Capability<&{FungibleToken.Receiver}>?): NonFungibleToken.NFT
+```
+Facilitates the purchase of the listing by providing the payment vault
+and the commission recipient capability if there is a non-zero commission for the given listing.
+Respective saleCuts are transferred to beneficiaries and funtion return underlying or listed NFT.
+
+---
+
+**fun `getDetails()`**
+
+```cadence
+fun getDetails(): ListingDetails
+```
+Fetches the details of the listings
+
+---
+
+**fun `getAllowedCommissionReceivers()`**
+
+```cadence
+fun getAllowedCommissionReceivers(): [Capability<&{FungibleToken.Receiver}>]?
+```
+Fetches the allowed marketplaces capabilities or commission receivers for the underlying listing.
+If it returns `nil` then commission is up to grab by anyone.
```

Comment:
@satyamakgec similar to previous comment, how does commission beneficiary get set?